stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/factory/StreamOfflinePluginFactory.kt

```diff
@@ -190,6 +198,21 @@ public class StreamOfflinePluginFactory(
         )
     }
 
+    private fun getDeleteMessageListenerDatabase(
+        chatClient: ChatClient,
+        statePlugin: StatePlugin,
+    ): DeleteMessageListener {
+        val deleteMessageListenerDatabase = DeleteMessageListenerDatabase(
+            clientState = chatClient.clientState,
+            messageRepository = chatClient.repositoryFacade,
+            userRepository = chatClient.repositoryFacade
```

Comment:
same here

Reply:
The `DeleteMessageListenerDatabase` constructor requires the concrete repository, it should be fine to provide the `RepositoryFacade` here, because it implement all of our `Repositories`.
We don't have a way to extract a concrete repository from our `RepositoryFacade`

Reply:
The thing is that @leandroBorgesFerreira has already decomposed `RepositoryFacade` calls into a set of separate repositories calls. That's why it has no sense to pass the `RepositoryFacade` instance because otherwise, it will lead to duplicate calls to DB.

Reply:
🤔  I think it won't make duplicated class to DB. In the end, we are just calling RepositoryFacade in different alias. I need to use the Repositories instead of the RepositoryFacade because I need interfaces that are inside LLC module instead of a implementation that live inside a more specific module